### PR TITLE
Add StyleCop.Analyzers to projects

### PR DESCRIPTION
**Pull request by Nihlus**
This PR adds the [StyleCop.Analyzers](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) package to all projects in the solution, enabling automated style enforcement as a part of the build process.

Initially, all rules were disabled, and were incrementally enabled. Only those rules which did not produce any errors on the unchanged OpenTK codebase were left enabled, and the rest were disabled. This sets up a nice base to work off of in the future. A rule can be enabled and its errors corrected, whereupon it can be submitted as a complete pull request.

At present, rules are not set to warn, as the sheer amount of violations slow down build times considerably. Furthermore, a number of rules do not have an official ruling on the maintainer's parts, and will require some discussion before they are implemented.

### CODE REVIEW

File: src/OpenTK/OpenTK.iOS.csproj [hunk 1]
```diff
@@ -24,6 +26,7 @@
     <ConsolePause>false</ConsolePause>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DocumentationFile>bin\Debug\iOS\OpenTK.xml</DocumentationFile>
+    xed
```
Comment (Frassle) [inline src/OpenTK/OpenTK.iOS.csproj]: What?

**Comment by Frassle** [comment]
So just to confirm, appveyor and travis will run these checks?

File: src/OpenTK/OpenTK.iOS.csproj (hunk 1 above)
Comment (Nihlus) [inline src/OpenTK/OpenTK.iOS.csproj]: Oops. I was editing it with xed, which is a notepad++ equivalent on Linux. I usually do it in the terminal, but I must've had another window focused. Removing.

**Comment by Nihlus** [comment]
@Frassle Yep! It runs as a part of the MSBuild process, so it'll run on AppVeyor, Travis, and the user's local computer when they build. It injects actual build errors into the compiler.

**Comment by varon** [comment]
This is really cool. Nice to see a net decrease of LOC as well.

**Pull request merged**